### PR TITLE
Final PR - Project update

### PR DESCRIPTION
**Pull request by Ambre-Thevenin**
PR with : 
- New Geocoding API and function to reduce API calls 
- Shops Module up and running
- Navigation module up and running
- Sparkle module with 2 arrays for Entertainment 
- CSS fixes and updates 
- Several fix commits

### CODE REVIEW

**Review by Scarabotte — APPROVED** [review]

**Review by AdeCisy — APPROVED** [review]
Verified : OK for me

**Review by Barbeu78 — DISMISSED** [review]

File: src/components/Nav.jsx [hunk 1]
```diff
@@ -13,45 +13,28 @@ function Nav() {
...
-          </a>
-        </li>
-      </ul>
-      <button className="navbar_burger" onClick={handleShowLinks}>
-        <span className="burger-nav"></span>
-      </button>
-      <button className="navbar_burger" onClick={handleShowLinks}>
-        <span className="burger-plus"></span>
-      </button>
+      <div className="container">
```
Comment (pelegrand) [inline src/components/Nav.jsx:16]: Use the same CSS method for all your files plain css or modules

**Review by pelegrand — CHANGES_REQUESTED** [review]

**Review by pelegrand — APPROVED** [review]

File: src/components/HourlyMeteo.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  useEffect(async () => {
+    const result = await axios.get(
+      `https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${long}&appid=${api_weather}&units=metric&lang=fr&exclude=minutely`,
+    );
+    let res = result.data;
+    setForecast(res.hourly);
+  }, [coordinates, destination]);
+
+  let forecastHours = forecast.slice(0, 8);
```
Comment (tyZulh) [inline src/components/HourlyMeteo.jsx:21]: Transform this let in a state and apply the slice inside a useEffect 
```suggestion
const [forcastHours, setForecastHours] = useState([]);

useEffect(() => {
  setForcastHours(forcast.slice(0, 8);
}, [forcast])
```

File: src/components/jokes.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
+
+import AstroJokes from './JokesArray';
+
+import styles from './jokes.module.css';
+
+function RandomJokes() {
+  const FirstToSeeJokes = 'De quoi avez-vous besoin ?';
+  const [textValue, setTextValue] = useState(FirstToSeeJokes);
+  const SingleJoke = [...AstroJokes];
```
Comment (tyZulh) [inline src/components/jokes.jsx:10]: Deprecated

**Review by tyZulh — CHANGES_REQUESTED** [review]

**Review by tyZulh — APPROVED** [review]

**Pull request merged**